crates/autopilot/src/run_loop.rs

```diff
@@ -552,7 +558,10 @@ pub fn solve_request(
                     class,
                     app_data: order.data.app_data,
                     signature: order.signature.clone(),
-                    fee_policies: fee_policies.get(&order.metadata.uid).unwrap_or_default(),
+                    fee_policies: fee_policies
+                        .get(&order.metadata.uid)
+                        .cloned()
+                        .unwrap_or_default(),
```

Comment:
Why aren't fee policies part of the `Auction` and auction specific order struct to begin with? 
Wouldn't it be reasonable to build the fee policies at the same time when we build the auction rather than putting it on the critical path of sending the solve request?

Reply:
Because then it would be a requirement to attach fee policies every time the order is created (because we use `model::Auction` directly instead of autopilot domain specific type) which means:
1. Fee policies have to be defined for both `orderbook` and `autopilot`
2. Fee policies are determined at the order creation time, so if `autopilot` and `orderbook` restart with new fee policies all existing orders have old policies and will be settled with old which seems weird, but maybe ok (so it depends on the expectation, it wouldn't be a bug but rather a feature).

IMO fee policies are strictly protocol related and should live in `autopilot` only. The proper way would be to move the fee policies building into autopilots`SolvableOrdersCache` where `SolvableOrdersCache` would have to return autopilot domain specific/enriched `Auction` type containing build fee policies but that requires significant refactor.

Reply:
But if you think we should expand model::Order I can try that and see how far I come

Reply:
No I agree with your assessment that this should be part of the SolvableOrdersCache (and that should return the domain specific type).

If this is too much work to change right now we can continue converting the shared model into the domain specific model here. However, I'd still prefer letting the `FeePolicyBuilder` or `FeeEngine` operate on a core domain model (rather than defining new types that we glue together in the runloop).

Do you think we can have FeeEngine expose a method `fn apply_fees(auction: domain::Auction) -> domain::Auction` that we call before invoking `solveRequest(domain::Auction)` and which then simply converts the domain object into the dto?

So we would do something like
1.  solvable_orders -> shared::Auction
2. here: convert shared::Auction -> domain::Auction
3. apply_fee on domain::Auction
4. in solve_request domain::Auction -> dto